app/components/avo/fields/common/files/controls_component.rb

```diff
@@ -4,16 +4,13 @@ class Avo::Fields::Common::Files::ControlsComponent < Avo::BaseComponent
   include Avo::ApplicationHelper
   include Avo::Fields::Concerns::FileAuthorization
 
-  attr_reader :file, :field, :resource
-  delegate :id, to: :field
+  delegate :id, to: :@field
 
-  def initialize(field:, file:, resource:)
-    @field = field
-    @file = file
-    @resource = resource
-  end
+  prop :field, Avo::Fields::BaseField
+  prop :file, ActiveStorage::Attachment
```

Comment:
This breaks eager-loading on rails apps that dont have `active_storage/engine` required

Reply:
Thanks! We missed it before.
Fixed by https://github.com/avo-hq/avo/pull/3215/